pyanaconda/modules/boss/installation.py

```diff
@@ -230,3 +290,489 @@ def run(self):
+            configuration_queue.append_dbus_tasks(SECURITY, [
+                security_proxy.ConfigureFIPSWithTask()
+            ])
+
+            # Join a realm. This can run only after network
+            # is configured in the target system chroot.
+            configuration_queue.append_dbus_tasks(SECURITY, [
+                security_proxy.JoinRealmWithTask()
+            ])
+
+        # Calling zipl should be the last task on s390
+        configuration_queue.append_dbus_tasks(STORAGE, [
+            bootloader_proxy.FixZIPLBootloaderWithTask()
+        ])
+
+        # setup kexec reboot if requested
+        if flags.flags.kexec:
+            kexec_setup = TaskQueue(
+                "Kexec setup",
+                _("Setting up kexec"),
+                CATEGORY_BOOTLOADER
+            )
+            kexec_setup.append(Task(
+                "Setup kexec",
+                setup_kexec
+            ))
+            configuration_queue.append(kexec_setup)
+
+        # write anaconda related configs & kickstarts
+        write_configs = TaskQueue(
+            "Write configs and kickstarts",
+            _("Storing configuration files and kickstarts"),
+            CATEGORY_SYSTEM
+        )
+
+        # Write the kickstart file to the installed system (or, copy the input
+        # kickstart file over if one exists).
+        if conf.target.can_save_output_kickstart:
+            # write anaconda related configs & kickstarts
+            write_configs.append(Task("Store kickstarts", _writeKS_via_boss))
+        else:
+            # don't write the kickstart file to the installed system if this has
+            # been disabled by the nosave option
+            log.warning("Writing of the output kickstart to installed system has been disabled"
+                        " by the nosave option.")
+
+        # only add write_configs to the main queue if we actually store some kickstarts/configs
+        if write_configs.task_count:
+            configuration_queue.append(write_configs)
+
+        post_scripts = TaskQueue(
+            "Post installation scripts",
+            _("Running post-installation scripts"),
+            CATEGORY_SYSTEM
+        )
+        scripts_proxy = RUNTIME.get_proxy(SCRIPTS)
+        post_scripts.append_dbus_tasks(RUNTIME, [
+            scripts_proxy.RunScriptsWithTask(KS_SCRIPT_POST)
+        ])
+        configuration_queue.append(post_scripts)
+
+        finish_tasks = boss_proxy.FinishInstallationWithTasks()
+        configuration_queue.append_dbus_tasks(BOSS, finish_tasks)
+
+        return configuration_queue
+
+    def _prepare_installation(self):
+        """Perform an installation.  This method takes the ksdata as prepared by
+           the UI (the first hub, in graphical mode) and applies it to the disk.
+           The two main tasks for this are putting filesystems onto disks and
+           installing packages onto those filesystems.
+        """
+        payload = ActiveDBusPayload()
```

Comment:
Maybe I'd update the doc string so it says it creates queue with tasks for what is said in the current doc.